com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RenameOldAppsStep.java

```diff
@@ -0,0 +1,101 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+@Named("renameOldAppsStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class RenameOldAppsStep extends RenameAppsStep {
+
+    @Override
+    protected StepPhase executeStepInternal(ExecutionWrapper execution) {
+        DeployedMta deployedMta = StepsUtil.getDeployedMta(execution.getContext());
+        if (deployedMta == null) {
+            return StepPhase.DONE;
+        }
+
+        getStepLogger().debug("Renaming old applications");
+
+        CloudControllerClient client = execution.getControllerClient();
```

Comment:
This client is created here, but used 10 lines below. You should aim to minimize the scope of variables as much as possible:
http://www.corejavaguru.com/effective-java/items/45